pkg/network/telemetry/ebpf_telemetry.go

```diff
@@ -377,3 +377,13 @@ func ActivateBPFTelemetry(m *manager.Manager, undefinedProbes []manager.ProbeIde
 
 	return nil
 }
+
+func EBPFTelemetrySupported() bool {
+	kversion, err := kernel.HostVersion()
+	if err != nil {
+		log.Warn("could not determine the current kernel version. EBPF telemetry disabled.")
+		return false
+	}
+
+	return kversion >= kernel.VersionCode(4, 14, 0)
```

Comment:
Is this based on some feature in the kernel? If yes, then it may be better to check for that so that checks like this run correctly on platforms such as centos/redhat. Use `https://github.com/cilium/ebpf/tree/main/features` if possible?

Reply:
It is based on the presence of the 'lock xadd' BPF instruction support in the kernel.
I am not sure if 4.14 is exactly when it became available, however.

After [#18398](https://github.com/DataDog/datadog-agent/pull/18398) maybe we don't need the atomic instruction altogether..

Reply:
`BPF_XADD` has been in the instruction set since the beginning.

Reply:
The `LOCK` prefix was added later if I am not mistaken.

Reply:
The only significant change to atomics that I'm aware of was done in 5.12 https://github.com/torvalds/linux/commit/7064a7341a0d2fcfeff56be7e3917421fbb8b024

Reply:
`BPF_XADD | BPF_W | BPF_STX` is definitely supported, which is what you are doing with `asm.StoreXAdd(asm.R1, asm.R2, asm.Word)`

Reply:
Ah yes you're right. I revisited why we patched this instruction, and it was because the verifier on 4.4 kernels was not smart enough to track bounds when we wrote to the array containing telemetry counters. So nothing to do with the atomics.

I can try and do a 'verifier bisect' to see what is the version we need exactly.

Reply:
Any update on the real version required here?

Reply:
Yes. Turns out 4.14 is the first version on which we can pass the verifier.